app/views/govuk_publishing_components/components/_summary_list.html.erb

```diff
@@ -0,0 +1,67 @@
+<%
+  id ||= nil
+  title ||= nil
+  edit ||= {}
+  items ||= []
+  block ||= yield
+%>
+<% if title || items.any? %>
+  <%= tag.div class: "gem-c-summary-list", id: id do %>
+    <% if title %>
+      <%= tag.h3 title, class: "govuk-heading-m" %>
+      <% if edit.any? %>
+        <%= tag.ul class: "govuk-summary-list__actions-list" do %>
+          <%= tag.li class: "govuk-summary-list__actions-list-item" do %>
+            <%= link_to edit.fetch(:href),
+                      class: "govuk-link gem-c-summary-list__edit-section-link",
+                      title: "Edit #{title}",
+                      data: edit.fetch(:data_attributes, {}) do %>
+               Edit<%= tag.span title, class: "govuk-visually-hidden" %>
+            <% end %>
+          <% end %>
+        <% end %>
+      <% end %>
+    <% end %>
+
+    <% if items.any? %>
+      <%= tag.dl class: "govuk-summary-list" do %>
+        <% items.each do |item| %>
+          <%= tag.div class: "govuk-summary-list__row" do %>
+
+            <%= tag.dt item[:field], class: "govuk-summary-list__key" %>
+            <%= tag.dd item[:value], class: "govuk-summary-list__value" %>
+
+            <% if item.fetch(:edit, {}).any? || item.fetch(:delete, {}).any? %>
+              <%= tag.dd class: "govuk-summary-list__actions" do %>
+                <%= tag.ul class: "govuk-summary-list__actions-list" do %>
+                  <% if item.fetch(:edit, {}).any? %>
+                    <%= tag.li class: "govuk-summary-list__actions-list-item" do %>
+                      <%= link_to item[:edit].fetch(:href),
+                                  class: "govuk-link",
+                                  title: "Edit #{item[:field]}",
+                                  data: item[:edit].fetch(:data_attributes, {}) do %>
+                        Edit<%= tag.span item[:field], class: "govuk-visually-hidden" %>
+                      <% end %>
+                    <% end %>
+                  <% end %>
+                  <% if item.fetch(:delete, {}).any? %>
+                    <%= tag.li class: "govuk-summary-list__actions-list-item" do %>
+                      <%= link_to item[:delete].fetch(:href),
+                                  class: "govuk-link",
+                                  title: "Delete #{item[:field]}",
+                                  data: item[:edit].fetch(:data_attributes, {}) do %>
+                        Delete<%= tag.span item[:field], class: "govuk-visually-hidden" %>
+                      <% end %>
+                    <% end %>
+                  <% end %>
+                <% end %>
+              <% end %>
+            <% end %>
+          <% end %>
+        <% end %>
+      <% end %>
+    <% end %>
```

Comment:
I wonder if we can cut down any of the nesting by doing more conditional checks up front and less 'wrapping' deeper inside the nest?

Reply:
Hmm.. it's not great in terms of nesting, but this is the level of granularity we want the mark-up to be, as there are lots of optional bits where we don't want to render markup unless we have a value for it. Open to suggestions if you see a way to make the code more readable while rendering things optionally.

Reply:
I suppose you could make a helper `list-item` component, as the list items all look the same apart from the title prefix ("Delete", "Edit" etc)

But also wouldn't want to go for 'the wrong abstraction'! Happy to leave with you.